v0/destinations/ga/deleteUsers.js

```diff
@@ -28,37 +28,64 @@ const userDeletionHandler = async (userAttributes, config, rudderDestInfo) => {
       .setStatus(500)
       .build();
   }
-  await Promise.all(
-    userAttributes.map(async userAttribute => {
-      if (!userAttribute.userId) {
-        throw new ErrorBuilder()
-          .setMessage("User id for deletion not present")
-          .setStatus(400)
-          .build();
+  const requests = userAttributes.map(userAttribute => {
+    if (!userAttribute.userId) {
+      throw new ErrorBuilder()
+        .setMessage("User id for deletion not present")
+        .setStatus(400)
+        .build();
+    }
+    // Reference for building userDeletionRequest
+    // Ref: https://developers.google.com/analytics/devguides/config/userdeletion/v3/reference/userDeletion/userDeletionRequest#resource
+    const reqBody = {
+      kind: "analytics#userDeletionRequest",
+      id: {
+        type: "USER_ID",
+        userId: userAttribute.userId
       }
-      // Reference for building userDeletionRequest
-      // Ref: https://developers.google.com/analytics/devguides/config/userdeletion/v3/reference/userDeletion/userDeletionRequest#resource
-      const reqBody = {
-        kind: "analytics#userDeletionRequest",
-        id: {
-          type: "USER_ID",
-          userId: userAttribute.userId
+    };
+    // TODO: Check with team if this condition needs to be handled
+    if (config.useNativeSDK) {
+      reqBody.propertyId = config.trackingID;
+    } else {
+      reqBody.webPropertyId = config.trackingID;
+    }
+    const headers = {
+      Authorization: `Bearer ${secret?.access_token}`,
+      Accept: "application/json",
+      "Content-Type": "application/json"
+    };
+    return {
+      body: reqBody,
+      headers
+    };
+  });
+  return requests;
+};
+
+/**
+ * This function will help to delete the users one by one from the userAttributes array.
+ *
+ * @param {*} userAttributes Array of objects with userId, emaail and phone
+ * @param {*} config Destination.Config provided in dashboard
+ * @param {Record<string, any> | undefined} rudderDestInfo contains information about the authorisation details to successfully send deletion request
+ * @returns {Array<{ body: any, headers: any }>}
+ */
+const userDeletionHandler = async (userAttributes, config, rudderDestInfo) => {
+  const userDeleteRequests = prepareDeleteRequest(
```

Comment:
can we make prepareDeleteRequest too to parallel processing ?
 await Promise.all(
    userAttributes.map(async user => {const request = await prepareDeleteRequest(user, config, rudderDestInfo); return request;})
    );

Reply:
If that is the intended change, what is the intended difference between `userDeletionHandler` and `prepareDeleteRequest` ?

My understanding was that `prepareDeleteRequest` is like the request formation or preparation stage.
The actual requests(using http client like axios) will be made in `userDeletionHandler`
Can you elaborate on this ?

Reply:
As we don't have batching for UA, other optimization is to do as much parallel processing as possible - so `prepareDeleteRequest` is also one good candidate

Reply:
The async version on `prepareDeleteRequest` is performing a little poorly than the current implemented one